activerecord/lib/active_record/relation/where_clause.rb

```diff
@@ -39,10 +39,16 @@ def or(other)
         if left.empty? || right.empty?
           common
         else
-          or_clause = WhereClause.new(
-            [left.ast.or(right.ast)],
-          )
-          common + or_clause
+          left = left.ast
+          left = left.expr if left.is_a?(Arel::Nodes::Grouping)
+
+          right = right.ast
+          right = right.expr if right.is_a?(Arel::Nodes::Grouping)
```

Comment:
This is a safe transformation because of the fact that SQL `OR` has lower precedence than `AND`, yes?

Reply:
I think yes.